moveit_setup_assistant/src/widgets/perception_widget.cpp

```diff
@@ -0,0 +1,345 @@
+  point_cloud_form_layout->addRow("Padding Scale:", padding_scale_field_);
+
+  // Filtered Cloud Topic
+  filtered_cloud_topic_field_ = new QLineEdit(this);
+  filtered_cloud_topic_field_->setMaximumWidth(400);
+  point_cloud_form_layout->addRow("Filtered Cloud Topic:", filtered_cloud_topic_field_);
+
+  // Max Update Rate
+  max_update_rate_field_ = new QLineEdit(this);
+  max_update_rate_field_->setMaximumWidth(400);
+  point_cloud_form_layout->addRow("Max Update Rate:", max_update_rate_field_);
+
+  // Piont Cloud form layout
+  point_cloud_group_->setLayout(point_cloud_form_layout);
+  layout->addWidget(point_cloud_group_);
+
+  // Depth map group --------------------------------------------
+  depth_map_group_ = new QGroupBox("Depth Map");
+
+  // Depth Map form layout
+  QFormLayout* depth_map_form_layout = new QFormLayout();
+  depth_map_form_layout->setContentsMargins(0, 15, 0, 15);
+
+  // Image Topic
+  image_topic_field_ = new QLineEdit(this);
+  image_topic_field_->setMaximumWidth(400);
+  depth_map_form_layout->addRow("Image Topic:", image_topic_field_);
+
+  // Queue Size
+  queue_size_field_ = new QLineEdit(this);
+  queue_size_field_->setMaximumWidth(400);
+  depth_map_form_layout->addRow("Queue Size:", queue_size_field_);
+
+  // Near Clipping Plane Distance
+  near_clipping_field_ = new QLineEdit(this);
+  near_clipping_field_->setMaximumWidth(400);
+  depth_map_form_layout->addRow("Near Clipping Plane Distance:", near_clipping_field_);
+
+  // Far Clipping Plane Distance
+  far_clipping_field_ = new QLineEdit(this);
+  far_clipping_field_->setMaximumWidth(400);
+  depth_map_form_layout->addRow("Far Clipping Plane Distance:", far_clipping_field_);
+
+  // Shadow Threshold
+  shadow_threshold_field_ = new QLineEdit(this);
+  shadow_threshold_field_->setMaximumWidth(400);
+  depth_map_form_layout->addRow("Shadow Threshold:", shadow_threshold_field_);
+
+  // Padding Offset
+  depth_padding_offset_field_ = new QLineEdit(this);
+  depth_padding_offset_field_->setMaximumWidth(400);
+  depth_map_form_layout->addRow("Padding Offset:", depth_padding_offset_field_);
+
+  // Padding Scale
+  depth_padding_scale_field_ = new QLineEdit(this);
+  depth_padding_scale_field_->setMaximumWidth(400);
+  depth_map_form_layout->addRow("Padding Scale:", depth_padding_scale_field_);
+
+  // Filtered Cloud Topic
+  depth_filtered_cloud_topic_field_ = new QLineEdit(this);
+  depth_filtered_cloud_topic_field_->setMaximumWidth(400);
+  depth_map_form_layout->addRow("Filtered Cloud Topic:", depth_filtered_cloud_topic_field_);
+
+  // Filtered Cloud Topic
+  depth_max_update_rate_field_ = new QLineEdit(this);
+  depth_max_update_rate_field_->setMaximumWidth(400);
+  depth_map_form_layout->addRow("Max Update Rate:", depth_max_update_rate_field_);
+
+  depth_map_group_->setLayout(depth_map_form_layout);
+  layout->addWidget(depth_map_group_);
+
+  // Perception documentation page
+  QLabel* perception_documentation_hyperlink = new QLabel(this);
```

Comment:
move to top as we discussed please. or move to bottom but don't have it jump around :)